telegram/ext/_application.py

```diff
@@ -1251,30 +1251,43 @@ async def process_update(self, update: object) -> None:
             try:
                 for handler in handlers:
                     check = handler.check_update(update)  # Should the handler handle this update?
-                    if not (check is None or check is False):  # if yes,
-                        if not context:  # build a context if not already built
+                    if check is None or check is False:
```

Comment:
```suggestion
                    if not check:
```

Reply:
No, we can't change this. We explicitly document that `None` or `False` indicate that the handler should not trigger but anything else will be passed to `handle_update` https://docs.python-telegram-bot.org/en/v21.1.1/telegram.ext.basehandler.html#telegram.ext.BaseHandler.check_update. Users could be returning `[]` or some other custom object that evaulates to falsy.

Reply:
hmm I suspected that but didn't bother to check the docs, my bad.